core/src/glue.rs

```diff
@@ -27,28 +27,39 @@ impl<T, U: GStore<T> + GStoreMut<T>> Glue<T, U> {
         }
     }
 
-    pub async fn plan<Sql: AsRef<str>>(&self, sql: Sql) -> Result<Statement> {
+    pub async fn plan<Sql: AsRef<str>>(&self, sql: Sql) -> Result<Vec<Statement>> {
         let parsed = parse(sql)?;
-        let statement = translate(&parsed[0])?;
         let storage = self.storage.as_ref().unwrap();
 
-        plan(storage, statement).await
+        let mut statements = Vec::<Statement>::new();
+        let parsed_iter = parsed.iter();
+        for parsed in parsed_iter {
+            let translated = translate(parsed)?;
+            let planned = plan(storage, translated).await?;
+            statements.push(planned);
+        }
+        Ok(statements)
```

Comment:
```suggestion
        stream::iter(parsed)
            .map(|parsed| translate(&parsed))
            .then(|statement| async move { plan(storage, statement?).await })
            .try_collect()
            .await
```

we recommend fp style code
you can use `futures::Stream` to replace mutable variable
- add `futures::{TryStreamExt, StreamExt}`